### PR TITLE
ONNX Post-Training Quantization CamVid, Mapillary segmentation sample

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Introduce ONNX Post-Training Quantization CamVid, Mapillary segmentation sample

### Reason for changes

Extend support of PTQ of ONNX models

### Tests

Sanity test was added

### CODE REVIEW

File: tests/onnx/sanity_tests/test_segmentation_sanity_sample.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    onnx_output_model_path = str(tmp_path / model_name)
+
+    run(onnx_model_path, onnx_output_model_path, 'CamVid',
+        'none', batch_size=1, shuffle=True, num_init_samples=1,
+        input_shape=input_shape, ignored_scopes=None)
+
+    sess = rt.InferenceSession(onnx_output_model_path, providers=['OpenVINOExecutionProvider'])
+    _input = np.random.random(input_shape)
+    input_name = sess.get_inputs()[0].name
+    _ = sess.run([], {input_name: _input.astype(np.float32)})
```
Comment (vshampor) [inline tests/onnx/sanity_tests/test_segmentation_sanity_sample.py:76]: How long does this take?

**Review by vshampor — CHANGES_REQUESTED** [review]

File: tests/onnx/sanity_tests/test_segmentation_sanity_sample.py (hunk 1 above)
Comment (kshpv) [reply tests/onnx/sanity_tests/test_segmentation_sanity_sample.py:76]: Fast

File: examples/experimental/onnx/segmentation/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Segmentation sample
```
[the remark this answers is not in the document]
Comment (kshpv) [reply examples/experimental/onnx/segmentation/README.md]: Done

File: examples/experimental/onnx/semantic_segmentation/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Semantic Segmentation sample
+
+This sample shows an example of quantization of segmentation models. The used datasets are CamVid, Mapillary.
```
Comment (AlexKoff88) [inline examples/experimental/onnx/semantic_segmentation/README.md:3]: Please replace "This sample shows an example" rhyme with something, like "This example demonstrates quantization of ...". In this case, it is better to change the title "Semantic Segmentation Example"

Comment (AlexKoff88) [reply examples/experimental/onnx/semantic_segmentation/README.md:3]: I would even specify what kind of models, like  "This example demonstrates quantization of UNet and ICNet segmentation models on CamVid and Mapillary datasets."

File: examples/experimental/onnx/semantic_segmentation/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+Please, install the requirements for ONNX Post-Training Quantization of NNCF.
+
+Install requirements
+
+```
+pip install -r <nncf dir>/nncf/experimental/onnx/requirements.txt
+pip install -r <nncf dir>/examples/experimental/onnx/requirements.txt
+```
+
+## Getting the quantized model
```
Comment (AlexKoff88) [inline examples/experimental/onnx/semantic_segmentation/README.md:16]: "Getting the model" step is required to guide where to take it. This step can be called, like "Quantizing the model"

File: examples/experimental/onnx/semantic_segmentation/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+To run post-training quantization on your model you can use the following command.
+
+```
+python onnx_ptq_segmentation.py -m <ONNX model path> -o <quantized ONNX model path> --dataset_name <CamVid or Mapillary> --data <dataset path>
+```
+
+Also, you could specify some options of quantization, please, take a look at the argument description by using the
```
Comment (AlexKoff88) [inline examples/experimental/onnx/semantic_segmentation/README.md:24]: After the last discussion, I presume that there will be no extra options.

File: examples/experimental/onnx/semantic_segmentation/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+command:
+
+```
+python onnx_ptq_classification.py --help
+```
+
+## Measuring the accuracy of the original and quantized models
```
Comment (AlexKoff88) [inline examples/experimental/onnx/semantic_segmentation/README.md:31]: Validating the original and quantized models

File: examples/experimental/onnx/semantic_segmentation/README.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+## Results of Post-Training quantization of ONNX models
+
+| Model |  Original accuracy  | Quantized model accuracy |
+|:-----:|:-------------------:|:------------------------:|
+| ICNet |   67.88% (CamVid)   |      67.8% (CamVid)      |
+| UNet  |   71.95% (CamVid)   |     71.85% (CamVid)      |
+| UNet  |  56.24% (Mapillary) |     56.19% (Mapillary)   |
+
+## Measuring the performance of the original and quantized models
```
Comment (AlexKoff88) [inline examples/experimental/onnx/semantic_segmentation/README.md:53]: Benchmarking the original and quantized models

File: examples/experimental/onnx/semantic_segmentation/README.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+Also, you could specify some options of quantization, please, take a look at the argument description by using the
+command:
+
+```
+python onnx_ptq_classification.py --help
+```
+
+## Measuring the accuracy of the original and quantized models
+
+If you would like to compare the accuracy of the original model and quantized one, you could
```
Comment (AlexKoff88) [inline examples/experimental/onnx/semantic_segmentation/README.md:33]: I wonder why do we need AC at all if we have a correct implementation of data loader. We can create a simple validation function and use skylearn for metric computation (e.g. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html). In this case, it would be an end-to-end example.

**Comment by vinnamkim** [comment]
Closed by #1184.
